src/Raven.Studio/typescript/models/resources/info/resourcesInfo.ts

```diff
@@ -25,6 +25,14 @@ class resourcesInfo {
         this.initObservables();
     }
 
+    public addResource(resource: resourceInfo) {       
+        const resources = this.sortedResources();
+        resources.push(resource);
+
+        resources.sort((a, b) => a.name.toLowerCase().localeCompare(b.name.toLowerCase()));
```

Comment:
optionally we can use: https://lodash.com/docs/4.17.2#sortedIndexBy - beacuse we insert to sorted array, but you can leave it